File: agentpress/integrations/js/agentpress-sdk.mjs
// AgentPress zero-dependency JavaScript SDK for browser, Node, Deno, and agent runtimes.
export class AgentPress {
  constructor(baseUrl = 'https://barneywohl.github.io/agentpress/') {
    this.baseUrl = baseUrl.endsWith('/') ? baseUrl : baseUrl + '/';
  }
  url(path = '') { return new URL(path, this.baseUrl).toString(); }
  async fetchJson(path) {
    const r = await fetch(this.url(path), { headers: { 'accept': 'application/json' }});
    if (!r.ok) throw new Error(`AgentPress fetch failed ${r.status}: ${path}`);
    return await r.json();
  }
  async fetchText(path) {
    const r = await fetch(this.url(path));
    if (!r.ok) throw new Error(`AgentPress fetch failed ${r.status}: ${path}`);
    return await r.text();
  }
  manifest() { return this.fetchJson('.well-known/agentpress.json'); }
  ingestion() { return this.fetchJson('.well-known/ai-ingestion.json'); }
  locales() { return this.fetchJson('locales/locale-index.json'); }
  articles() { return this.fetchJson('agentpress/articles/article-index.json'); }
  hubDirectory() { return this.fetchJson('agentpress/hub/agent-directory.json'); }
  hashManifest() { return this.fetchJson('agentpress/hash-manifest.json'); }
  llms(locale = 'en') { return locale === 'en' ? this.fetchText('llms.txt') : this.fetchText(`locales/llms.${locale}.txt`); }
  async selfTest() {
    const checks = await Promise.allSettled([
      this.manifest(), this.ingestion(), this.locales(), this.articles(), this.hubDirectory(), this.hashManifest(), this.llms()
    ]);
    return { ok: checks.every(x => x.status === 'fulfilled'), checks: checks.map((x, i) => ({ index: i, status: x.status, reason: x.reason?.message })) };
  }
}
export default AgentPress;

File: agentpress/integrations/sdk/python/agentpress_sdk.py
"""Zero-dependency AgentPress SDK for Python agents."""
import json
from urllib.parse import urljoin
from urllib.request import Request, urlopen

class AgentPress:
    def __init__(self, base_url="https://barneywohl.github.io/agentpress/", timeout=20):
        self.base_url = base_url.rstrip("/") + "/"
        self.timeout = timeout
    def url(self, path=""):
        return urljoin(self.base_url, path)
    def fetch_text(self, path):
        req = Request(self.url(path), headers={"Accept":"text/plain, application/json"})
        with urlopen(req, timeout=self.timeout) as r:
            return r.read().decode("utf-8")
    def fetch_json(self, path):
        return json.loads(self.fetch_text(path))
    def manifest(self): return self.fetch_json(".well-known/agentpress.json")
    def tools(self): return self.fetch_json("agentpress/tools/agentpress-tools.json")
    def routes(self): return self.fetch_json("agentpress/routes/agent-routes.json")
    def marketplace(self): return self.fetch_json("agentpress/marketplace/marketplace-index.json")
    def proof_scoreboard(self): return self.fetch_json("agentpress/external-proofs/proof-scoreboard.json")
    def browser_smoke(self): return self.fetch_json("agentpress/evidence/browser-smoke.json")
    def self_test(self):
        checks=[]
        for name, path in [("manifest",".well-known/agentpress.json"),("tools","agentpress/tools/agentpress-tools.json"),("routes","agentpress/routes/agent-routes.json"),("marketplace","agentpress/marketplace/marketplace-index.json"),("llms","llms.txt")]:
            try:
                body=self.fetch_text(path); checks.append({"name":name,"path":path,"ok":bool(body),"bytes":len(body)})
            except Exception as e:
                checks.append({"name":name,"path":path,"ok":False,"error":str(e)})
        return {"ok": all(c.get("ok") for c in checks), "checks": checks}
